woocommerce/woocommerce-ios
# Dynamic Dashboard: Inbox messages card

Closes: #12684 
Also closes #12737 


## Description

This PR completes the UI and logic for the inbox messages card. Changes include:
- Updated the view model to sync inbox messages and load content from storage as the single source of truth. Using storage is necessary as dismissing notes will remove items from storage and we need to listen to these changes.
- Updated the UI of the dashboard card to display contents and handle loading and error states.
- Handled navigation to the Inbox list view.
- Customized the layout for the inbox note row when displayed on the dashboard to match the design.
- Triggered syncing inbox from `DashboardViewModel`. We'll need to optimize the syncing of all cards in a separate PR - for now inbox messages are synced when the feature flag for Dynamic Dashboard M2 is enabled.

## Testing instructions

- Enable the feature flag `dynamicDashboardM2` and build the app.
- Enable the Inbox card on the dashboard.
- Confirm that the card now displays the latest 3 items from inbox and the UI matches the design.
- Confirm that all actions on the inbox rows work correctly.
- Tap View all inbox messages and confirm that the inbox view is displayed.
- Disconnect the Internet connect and reload the card. Confirm that the error state is displayed.
- Tap the ellipsis button and dismiss the card. Confirm that event `dynamic_dashboard_hide_card_tapped` is tracked with type `inbox`.

## Screenshots

<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/bbcfbfcb-c7a7-4da3-b9ea-125e9d8870f1" width=320 />


---
